[Feature/lecture bookmark view]: 북마크api 의 lecture_id -> lecture_uuid 변경

209512 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #REQ-LECT-005, 006
- 작업 요약: 북마크api 의 lecture_id -> lecture_uuid 변경/ 북마크 manager 생성

## 📄 상세 내용
- [ ] 프론트 업무요청 전문
> 프론트엔드 3팀의 업무요청
> 현재 강의 api에서 uuid와 id가 모두 사용되고 있습니다
> uuid: 강의 get, 리뷰 get
> id: 북마크 post, delete 할 때의 lecture_id
> 
> 다만, 현재 api로는 각 강의의 int type id를 알아낼 수는 없기에 북마크 관련 api도 uuid로 수정을 부탁드립니다
> 
- [ ] 강의 북마크 managers.py 생성

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

209512 committed:
🐛 fix(lecture): 북마크 api/ lecture_id -> lecture_uuid 변경

[전문]
프론트엔드 3팀의 업무요청:
현재 강의 api에서 uuid와 id가 모두 사용되고 있습니다
uuid: 강의 get, 리뷰 get
id: 북마크 post, delete 할 때의 lecture_id

다만, 현재 api로는 각 강의의 int type id를 알아낼 수는 없기에 북마크 관련 api도 uuid로 수정을 부탁드립니다

209512 committed:
💡 chore(lecture): 다른 브랜치에서의 작업 파일 삭제

Meoyoug commented on apps/studies/tests/test_groups.py:
```diff
@@ -17,17 +19,42 @@ def setUp(self) -> None:
-            "start_at": "2027-11-01",  # 날짜 한참 뒤로 수정
-            "end_at": "2028-11-10",  # 동일
+            "start_at": "2026-11-25",  # 시작일 날짜 잘 확인해주세요 (2026년으로 변경)
+            "end_at": "2027-11-30",  # 종료일 또한 날짜 수정 필요 (위와 마찬가지)
```
이거 아마 날짜 밸리데이션하는거 있으면 해당일이 도래하면 또 그때맞춰서 코드변경을해줘야하는데
```
timezone.now() + timedelta(days=100) 
```

이런식으로 테스트가 실행되는 날짜에 맞춰 유동적이게 변경하시면 해결됩니다.

209512 committed:
♻️ refactor(lecture): 강의 북마크 manager 추가

LectureBookmarkQueryset 생성
- for_user, search method 생성

-> LectureBookmark, view에서 사용, 수정

209512 committed:
🎨 style(lecture): code formatting

209512 committed:
💡 chore(lecture): mypy type check

209512 committed:
✅ test(studies): test_groups.py 테스트 시간 유동적으로 변경

2joonkim reviewed (changes requested):
예지님 확인했고 코치님 리뷰사항 반영해서 적어주셨군용(감사합니다)
일단 리베이스가 조금 잘못된 거 같은데 내일 최신화 시켜서 꼬인거 해결하고 다시 pr 올려주시면
머지하도록 하겠습니다!

Meoyoug reviewed (approved):
커밋 꼬인거 제가 해결하고 푸시했어요 머지할게요~
